src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1512,6 +1512,21 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnSrpClientNotificatio
     case OT_ERROR_NONE: {
         ChipLogProgress(DeviceLayer, "OnSrpClientNotification: Last requested operation completed successfully");
 
+        if (aHostInfo)
+        {
+            if (aHostInfo->mState == OT_SRP_CLIENT_ITEM_STATE_REMOVED)
+            {
+                GenericThreadStackManagerImpl_OpenThread * threadStackInstance = static_cast<GenericThreadStackManagerImpl_OpenThread *>(aContext);
+
+                // Clear memory for removed host
+                char * hostname = threadStackInstance->Impl()->mSrpClient.mHostName;
+                memset(reinterpret_cast<void *>(hostname), 0, sizeof(hostname));
+
+                threadStackInstance->Impl()->mSrpClient.mIsInitialized = true;
+                threadStackInstance->Impl()->mSrpClient.mInitializedCallback(threadStackInstance->Impl()->mSrpClient.mCallbackContext, CHIP_NO_ERROR);
```

Comment:
Check if `mInitializedCallback` is set?

Reply:
I added also checking it in ClearSrpHost method